data_structures/src/transaction_factory.rs

```diff
@@ -5,57 +5,65 @@ use crate::{
     },
     error::TransactionError,
     transaction::{DRTransactionBody, VTTransactionBody},
-    utxo_pool::{OwnUnspentOutputsPool, UnspentOutputsPool, UtxoDiff, UtxoSelectionStrategy},
+    utxo_pool::{
+        NodeUtxos, OwnUnspentOutputsPool, UnspentOutputsPool, UtxoDiff, UtxoSelectionStrategy,
+    },
 };
 use std::{collections::HashSet, convert::TryFrom};
 
+pub trait UtxoFantasy {
```

Comment:
Add a short description, something like:

```suggestion
/// Abstraction used to create transactions using a UtxoSet
pub trait UtxoFantasy {
```